NTR: Add 3 pharmacological relations for small molecules

cthoyt opened the pull request:
Closes #371

## Motivation

For ChIRO and the Chemical Roles Graph (https://github.com/chemical-roles/chemical-roles), we need more detailed relationships to describe causal interactions between chemicals (material entities) and receptors (material entity).

Existing relationships (i.e., [directly negatively regulates activity of](http://purl.obolibrary.org/obo/RO_0002449) and [directly positively regulates activity of](http://purl.obolibrary.org/obo/RO_0002450)) are not detailed enough.

See similar issue about drug-target relationships in https://github.com/oborel/obo-relations/issues/144.

## What this PR does

This PR creates 3 new properties:

| LUID | Name | Parent | Example |
|----------:|-------|--------|-------|
| 0018027 |  is agonist of | [directly positively regulates activity of](http://purl.obolibrary.org/obo/RO_0002450) | mibolerone (CHEBI:34849) is agonist of androgen receptor (PR:P10275) |
| 0018028 | is inverse agonist of | [directly negatively regulates activity of](http://purl.obolibrary.org/obo/RO_0002449) |  pimavanserin (CHEBI:133017) is inverse agonist of HTR2A (PR:P28223) |
| 0018029 | is antagonist of | [directly negatively regulates activity of](http://purl.obolibrary.org/obo/RO_0002449) | tretinoin (CHEBI:15367) is antagonist of Nuclear receptor ROR-beta (PR:Q92753) |


Big thanks to Jie, Alexander, Darren, Bill, and everyone else at the 2022-11-01 RO meeting who gave feedback and helped improve this PR.

sierra-moxon commented:
Would this also set precedence for adding other, more specific mechanism relations?
e.g. [mechanisms in biolink](https://biolink.github.io/biolink-model/docs/causal_mechanism_qualifier_enum.html)

cthoyt commented:
I think it would be totally fantastic to have biolink-like properties standardized in RO as well!

deepakunni3 commented:
+1 to standardizing Biolink predicates/properties in RO 🚀

wdduncan commented:
Notes for RO meeting:
- add domain and ranges
- perhaps we need versions that are molecule to molecule, and molecule to process
- place properties under more general super-properties that have correct domains and ranges

balhoff commented:
@cthoyt maybe we could create some test GO-CAMs and then decide which nodes you would like these relations to go between. It also might help us to define any additional rules that would be useful.

cthoyt commented:
@wdduncan in the last few commits, I moved these relations under the [directly negatively regulates activity of](http://purl.obolibrary.org/obo/RO_0002449) and [directly positively regulates activity of](http://purl.obolibrary.org/obo/RO_0002450), which have clear domain and range annotations both being material entities. The description of these properties directly addresses the comments made by @addiehl to better notate that the existence of a relation `mibolerone (CHEBI:34849) is agonist of androgen receptor (PR:P10275)` implies that there is some activity of `PR:P10275` that is getting modulated. 

I found the `example usage` annotation and added some explicit examples for each both in the PR description and in the ontology.

I added seeAlso links to related Wikidata properties.

I improved the descriptions for each as well.

cthoyt commented:
@balhoff here's a first stab at the relations that might be in a GO CAM

| subject name | subject curie | relation name | relation curie | object name | object curie |
|---------------|--------------|---------------|---------------|---------------|---------------|
| tretinoin  | CHEBI:15367 | is antagonist of  | RO:0018029 | Nuclear receptor ROR-beta | PR:Q92753 | 
| Nuclear receptor ROR-beta | PR:Q92753 | has activity ?? | RO:??? | activity nuclear receptor activity | [GO:0004879](http://amigo.geneontology.org/amigo/term/GO%3A0004879) |
| tretinoin | CHEBI:15367 | is small molecule inhibitor of | RO:0012006 | activity nuclear receptor activity | [GO:0004879](http://amigo.geneontology.org/amigo/term/GO%3A0004879) |

wdduncan commented:
Thanks @cthoyt !
This area is out of my domain of expertise. So, I am going to defer to those more knowledge than me about whether this PR is read to be merged :)

balhoff commented:
> @balhoff here's a first stab at the relations that might be in a GO CAM

@cthoyt great, thanks! For your 'has activity' cell, you would use [enables](http://purl.obolibrary.org/obo/RO_0002327). Should there be a way to infer 'is antagonist of'? For example:

`'is small molecule inhibitor of' o 'enabled by' -> 'is antagonist of'`

Or is there not enough information there to distinguish 'is antagonist of' from 'is inverse agonist of'?

Similar question for [is small molecule activator of](http://purl.obolibrary.org/obo/RO_0012005) and 'is agonist of'.

cthoyt commented:
@balhoff thank you for the feedback! I don't think we can make this super general as the difference between antagonism and inverse agonism is based on the way that the chemical binds to the receptor. Also, not all `'is small molecule inhibitor of' o 'enabled by' ` should be ` 'is antagonist of'`. This should only be true if only if we restrict the range of the `enabled by` to be *receptor activity* (GO:0004872) and we say something specific about the mechanism of bindin (seems a bit complicated for this first pass)

sierra-moxon commented:
In Biolink we define `antagonism` as "A causal mechanism in which the effector binds to a receptor and prevents activation by an agonist through competing for the binding site."  

And `inverse agonism` in Biolink is "A causal mechanism in which the effector binds to the same receptor-binding site as an agonist and antagonizes its effects, often exerting the opposite effect of the agonist by suppressing spontaneous receptor signaling."  

Do these definitions reflect your understanding of the difference @cthoyt?

cthoyt commented:
Yeah that’s basically what I wrote, but we can also update the definitions if you think that would be more clear

wdduncan commented:
@balhoff @sierra-moxon @addiehl   
Am I correct in assuming that you all have approved this PR, and it is ready to be merged?
